### PR TITLE
Update the release build

### PR DESCRIPTION
**Pull request by daxian-dbw**
# PR Summary

Update the release build:

- Use a new builder pool, which has the latest PowerShell 7.0.3 available.
- Remove catalog related steps, as the catalog file is unnecessary to PowerShell Gallery now.
- Update the signing xml file to sign the new "Polyfiller" assemblies.

## PR Checklist

- [x] PR has a meaningful title
    - Use the present tense and imperative mood when describing your changes
- [x] Summarized changes
- [ ] Make sure you've added one or more new tests
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] Documentation needed at [PowerShell-Docs](https://github.com/MicrosoftDocs/PowerShell-Docs)
        - [ ] Doc Issue filed: 

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/PowerShell/PSReadLine/pull/1930)

### CODE REVIEW

**Comment by daxian-dbw** [comment]
This change has been verified with the Azure DevOps build https://dev.azure.com/mscodehub/PSReadLine/_build/results?buildId=158037&view=results

File: .vsts-ci/releaseBuild.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ resources:
 
 jobs:
 - job: build_windows
-  pool: Package ES Lab A
+  pool: Package ES CodeHub Lab E
```
Comment (adityapatwardhan) [inline .vsts-ci/releaseBuild.yml:20]: This should be the standard pool `Package ES Standard Build`. Did you have problems using it? It seems you had it in one of your previous commits

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: .vsts-ci/releaseBuild.yml (hunk 1 above)
Comment (daxian-dbw) [reply .vsts-ci/releaseBuild.yml:20]: Yes, the build didn't start when using `Package ES Standard Build`. It says something like "it doesn't exist or I don't have permission". I copied this one from Paul's SecretManagement release build.

Comment (daxian-dbw) [reply .vsts-ci/releaseBuild.yml:20]: I just sent the email to request adding the agent pool to PSReadLine. I will update the pool name once the agent pool is added and my access is clear.
Before that, I would like to go with "Package ES CodeHub Lab E", as I need to release 2.2.0-beta1 today or tomorrow.

File: .vsts-ci/sign-module-files.xml [hunk 2]
```diff
@@ -10,5 +10,7 @@
 
     <file src="__INPATHROOT__\Microsoft.PowerShell.PSReadLine2.dll" signType="AuthenticodeFormer" dest="__OUTPATHROOT__\Microsoft.PowerShell.PSReadLine2.dll" />
     <file src="__INPATHROOT__\System.Runtime.InteropServices.RuntimeInformation.dll" signType="AuthenticodeFormer" dest="__OUTPATHROOT__\System.Runtime.InteropServices.RuntimeInformation.dll" />
```
Comment (adityapatwardhan) [inline .vsts-ci/sign-module-files.xml]: Does this come from .NET? Shouldn't this be already signed.

Comment (daxian-dbw) [reply .vsts-ci/sign-module-files.xml]: Yes, this comes from .NET, and yes it's already signed. I will skip this one from the signing.

**Review by adityapatwardhan — APPROVED** [review]
Approved when the pool is changed.

**Review by SteveL-MSFT — APPROVED** [review]

**Pull request merged**